var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -85,6 +85,9 @@ class Silo(AutotoolsPackage):
     conflicts("+hzip", when="@4.10.2-bsd,4.11-bsd")
     conflicts("+fpzip", when="@4.10.2-bsd,4.11-bsd")
 
+    # silo 4.11 doesn't compile with gcc 11.1
```

Comment:
Actually it does build, but we have an issue with the when condition of a patch not triggering

Reply:
I see from the CI that it also fails with the same errors I saw with gcc11.1 with gcc9.4. Do you know when this will be resolved? I submitted a PR for visit and the CI went and built silo 4.11 with gcc 11.1 on some system and failed. I put a restriction on the silo version to 4.10 in the visit spack package and was told that the correct fix was to make the fix in this PR. I would like to get my visit spack changes in so that I can meet the sustainability criteria of an ECP KPP3 I am submitting for visit. It seems like I should go back to the fix I initially proposed in the visit spack package.

How should I proceed?